mail/util.py

```diff
@@ -0,0 +1,62 @@
+# taken (and _very_ slightly adjusted)
+# from https://docs.aws.amazon.com/ses/latest/dg/smtp-credentials.html
+
+import base64
+import hashlib
+import hmac
+
+import ctx
+
+SMTP_REGIONS = [
+    'us-east-2',       # US East (Ohio)
+    'us-east-1',       # US East (N. Virginia)
+    'us-west-2',       # US West (Oregon)
+    'ap-south-1',      # Asia Pacific (Mumbai)
+    'ap-northeast-2',  # Asia Pacific (Seoul)
+    'ap-southeast-1',  # Asia Pacific (Singapore)
+    'ap-southeast-2',  # Asia Pacific (Sydney)
+    'ap-northeast-1',  # Asia Pacific (Tokyo)
+    'ca-central-1',    # Canada (Central)
+    'eu-central-1',    # Europe (Frankfurt)
+    'eu-west-1',       # Europe (Ireland)
+    'eu-west-2',       # Europe (London)
+    'sa-east-1',       # South America (Sao Paulo)
+    'us-gov-west-1',   # AWS GovCloud (US)
+]
+
+# These values are required to calculate the signature. Do not change them.
+DATE = "11111111"
+SERVICE = "ses"
+MESSAGE = "SendRawEmail"
+TERMINAL = "aws4_request"
+VERSION = 0x04
+
+
+def _sign(key, msg):
+    return hmac.new(key, msg.encode('utf-8'), hashlib.sha256).digest()
+
+
+def calculate_ses_key(secret_access_key: str, region:str) -> str:
+    if region not in SMTP_REGIONS:
+        raise ValueError(f"The {region} Region doesn't have an SMTP endpoint.")
```

Comment:
I would rather either calculate this dynamically, or mv this to cfg, or drop this check (it will fail during runtime + I do not think we will see a lot of additional usage, so making it more convenient to learn about supported regions - which should be documented by ses anyhow) is not much of a concern)

adding this as a comment is fine. however, I do not at all like hardcoding those regions, as I assume they might change in the future